Upgrade Protractor dependency to include v4

henrahmagix opened the pull request:
This replaces grunt-protractor-runner with a shell command so any
Protractor version later than the currently-supported v2.2.0 can be
used. (See also https://github.com/teerapap/grunt-protractor-runner/pull/164)

Also I don't think Protractor needs to be installed as
a dependency, so this moves it to a peer-dependency whilst also
including it as a dev-dependency so it can be run in the tests.

henrahmagix commented:
Ah, looks like there's an issue using node 0.10. I'll look into that shortly =)

henrahmagix commented:
I forgot to add: this would mean all tests will run in the latest Protractor, so issues with testability will show up sooner. You could also use [Travis cron jobs](https://docs.travis-ci.com/user/cron-jobs/) to periodically run the build to detect errors with new versions of Protractor.

alfonso-presa commented:
Thanks for this. I've just pushed un update to node 4 (as LTS is now there, is has all the sense to stop mantaining node 0.X), and also I've detected some race conditions happening with this PR (eventualy failing the last e2e test in chrome) so I've patched it in master. Please rebase against current master.

henrahmagix commented:
@alfonso-presa Ah cool! Ok I will rebase. What do you think of 97c30bb to allow support for v4, v5, and v6? I can remove v0.10 and v0.12 from the list now.

alfonso-presa commented:
Yeah, that looks great.

henrahmagix commented:
I've put v0.10 back, with v0.12, to see if that will work. I've also changed 4.0 to 4 so the latest will be used. A second commit has removed v0.x support.

Also, I've removed the `before_script` run of updating the webdriver, since that now happens in the `grunt test` command =)

alfonso-presa commented:
Please can you rebase again against master? I'm trying to make the PRs green in travis. Looks like sauce-connect addon is disabled for PRs so I've added a grunt plugin.

Sorry for the inconvenience.

henrahmagix commented:
No problem, there's no inconvenience =)

alfonso-presa commented on tasks/aliases.yaml:
```diff
@@ -5,6 +5,7 @@ test:
   - connect:serve
   - sauce_connect
   - protractor
```
I think protractor should be removed as grunt-protractor-runner is not installed now

henrahmagix replied:
Woops, I thought I had, my bad

alfonso-presa replied:
I think you did, but as I also modified aliases.yaml in master, probably it has get back again during merge. Sorry!

henrahmagix replied:
It's no worry! I clearly didn't pay full attention when resolving conflicts in rebasing =D

alfonso-presa commented:
Great. This PR is now green! :-)

henrahmagix commented:
Woo! Thanks!
